optaplanner-core/src/main/java/org/optaplanner/core/impl/io/jaxb/SolverConfigIO.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *  
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *  
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.io.jaxb;
+
+import java.io.Reader;
+import java.io.Writer;
+
+import org.optaplanner.core.config.solver.SolverConfig;
+
+public class SolverConfigIO {
```

Comment:
I prefer composition to avoid leaking the lower-level methods into code that uses the `SolverConfigIO`. Code, that ideally should know nothing about the magic with namespaces (and should not even have the possibility to call the other methods).

Let me see if I can make the other methods protected to change the composition for inheritance.

Reply:
Actually, after thinking about this further, inheritance does not make much sense here:

`SolverConfigIO` would redefine `read` and `write` methods of `JaxbIO`, completely changing their behavior. It would mean a violation of LSP, as the supertype is no longer replaceable by its subtype.

If we want to standardize the method signatures across IO implementation classes, it calls rather for an interface. But since we don't need polymorphism of the IO implementations, it sounds like an overengineering.

Reply:
In that case, I really don't understand why you're using `JaxbIO` at all. If you need parts of it, but don't want to inherit them, then those parts should be in some util class.